packages/x-components/src/x-modules/tagging/components/tagging.vue

```diff
@@ -0,0 +1,128 @@
+<script lang="ts">
+  import Vue from 'vue';
+  import { Component, Inject, Prop } from 'vue-property-decorator';
+  import { XEmit } from '../../../components/decorators/bus.decorators';
+  import { xComponentMixin } from '../../../components/x-component.mixin';
+  import { SnippetConfig } from '../../../x-installer/api/api.types';
+  import { taggingXModule } from '../x-module';
+
+  @Component({
+    mixins: [xComponentMixin(taggingXModule)]
+  })
+  export default class Tagging extends Vue {
+    /**
+     * It injects {@link SnippetConfig} provided by an ancestor as snippetConfig.
+     *
+     * @internal
+     */
+    @Inject('snippetConfig')
+    protected snippetConfig?: SnippetConfig;
+
+    /**
+     * The session TTL in milliseconds.
+     *
+     * @internal
+     */
+    @Prop()
+    @XEmit('SessionDurationProvided')
+    public sessionTTLMs: number | undefined;
+
+    /**
+     * The debounce time in milliseconds to track the query.
+     *
+     * @internal
+     */
+    @Prop()
+    @XEmit('QueryTaggingDebounceProvided')
+    public queryTaggingDebounceMs: number | undefined;
+
+    /**
+     * The consent to be emitted.
+     *
+     * @public
+     */
+    @Prop()
+    protected consent?: boolean;
+
+    /**
+     * The active consent, selected from the `consent` prop and the `snippetConfig.consent`
+     * property. False by default.
+     *
+     * @returns A boolean that represents if the consent is accepted or not.
```

Comment:
I would like to have a `@remaks` explaining that if every consent is undefined we'l return false.